src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -124,7 +158,8 @@ public virtual void EvaluatePreflightRequest(HttpContext context, CorsPolicy pol
 
                 if (!found)
                 {
-                    return;
+                    _logger?.PolicyFailure($"Request method {accessControlRequestMethod} not allowed in CORS policy.");
```

Comment:
I think it makes sense to give each of these policy failures it's own event id and all the ceremony that comes with that. You probably also want a single "policy failed" message that doesn't provide any detail.